### PR TITLE
feat(approval): fork-aware contextual exemption and automatic fork sync

### LINKED ISSUE #50: feat: add a sessionStart hook that ensures the checked out repo is in sync
If a `REPO_URL` is defined and the repo is a fork, we need to ensure that the checked-out repo is up to date with the upstream repo. If it's not, sync them, and ensure we push the latest commits to the fork's remote origin on GitHub.

### LINKED ISSUE #51: bug: valid gh api calls are triggering approvals when they are contextually safe
## Summary

When updating a GitHub issue comment via `gh api`, the command pattern used was incorrect:

```bash
gh api repos/perezd/claudetainer/issues/comments/ID -X PATCH --body-file /tmp/file.md -F "body=@/tmp/file.md"
```

This mixes `--body-file` (which is a flag for `gh issue comment`, not `gh api`) with `-F` (which sends form fields). The `gh api` command doesn't support `--body-file`.

## Context

This occurred while executing the Issue-Driven Workflow for #49. The implementation plan comment needed its checkboxes updated from `- [ ]` to `- [x]` after task completion. The CLAUDE.md CLI Best Practices section recommends `--body-file` for `gh` commands, but that guidance applies to `gh issue comment` and `gh pr create`, not `gh api`.

## Correct pattern for `gh api` comment updates

Use `--input` with a JSON payload created from the markdown file:

```bash
# Step 1: Convert markdown to JSON payload
jq -Rs '{body: .}' /tmp/file.md > /tmp/file.json

# Step 2: Send via --input (file-based, no inline strings)
gh api repos/OWNER/REPO/issues/comments/ID -X PATCH --input /tmp/file.json
```

This keeps content in files (consistent with the `--body-file` preference for other `gh` commands) and avoids inline body strings that trigger the command approval classifier.

## Suggested fix

Add a note to CLAUDE.md's CLI Best Practices section clarifying that for `gh api` updates, use `jq -Rs '{body: .}' file.md > payload.json` + `gh api ... --input payload.json` as the file-based equivalent of `--body-file`.

### PR DESCRIPTION
**Pull request by limbibot**
## Summary

Fixes #51. Implements #50.

The contextual `gh` command exemption (introduced in #36) only recognized `gh api` commands targeting the fork's origin URL. When the container clones a fork, commands targeting the upstream parent repo (e.g., `gh pr create --repo perezd/claudetainer`) were not recognized as contextually safe and triggered unnecessary approval prompts.

This PR:
- **Adds upstream parent URL to the boot-time snapshot** — the entrypoint detects forks via `gh repo view` and appends the upstream URL to `/tmp/approval/git-remote-urls.txt`
- **Extends contextual exemption to all mutating `gh` subcommands** — `gh issue comment/create/edit/close/reopen`, `gh pr create/comment/edit/merge/close/review`, `gh repo sync`, plus existing `gh api` support
- **Adds `ALWAYS_ESCALATE_HOT_WORDS`** — high-consequence operations (`gh pr merge`, `gh pr close`, `gh issue close`) always reach Haiku regardless of repo relationship
- **Adds `gh repo sync` parser** — validates both target fork and `--source` repo against the snapshot; `--force` is Tier 1 hard-blocked
- **Implements automatic fork sync** — `sync-fork.sh` runs at session start and before worktree creation via hooks, syncing the fork's main with upstream

## Layer-Impact Assessment

| Layer | Affected | Why |
|-------|----------|-----|
| Container Hardening | Yes | Dockerfile adds `sync-fork.sh` copy and chmod |
| Network Isolation | No | No network config changes |
| Command Approval | Yes | New hot words, Tier 1 block, parser, always-escalate set, extended contextual exemption |

## Panel Review

Full panel review completed with 4 core experts across 2 rounds. Key findings addressed:
- **Round 1 HIGH:** sync-fork.sh originally read from live git state (manipulable by claude user) — redesigned to read from immutable boot-time snapshot
- **Round 1 HIGH:** `--source=value` equals form in `gh repo sync` silently dropped source validation — fixed with equals-form parsing
- All experts approved in Round 2 with no remaining concerns

## Test Plan

- [x] All 314 approval classifier tests pass (including ~35 new tests)
- [x] Prettier formatting check passes
- [x] `parseGhRepoSyncTarget` tested for positional args, `--source` space/equals forms, `--branch` flag, edge cases
- [x] `ALWAYS_ESCALATE_HOT_WORDS` integration tested — `gh pr merge`, `gh pr close`, `gh issue close` always escalate
- [x] Tier 1 block for `gh repo sync --force` tested
- [x] All 12 new hot words have Tier 2 escalation tests
- [x] `isContextualGhCommand` integration tests for sync commands with related/unrelated repos

Closes #51
Closes #50

### CODE REVIEW

File: scripts/entrypoint.sh [hunk 1]
```diff
@@ -217,6 +217,34 @@ if [[ -n "${REPO_URL:-}" ]]; then
...
+  (
+    set +e
+    # Extract origin NWO from remote URL
+    origin_url=$(git -C /workspace/repo remote get-url origin 2>/dev/null)
+    repo_nwo=$(echo "$origin_url" | sed -E 's#.*/([^/]+/[^/.]+?)(\.git)?$#\1#')
+    if [[ -n "$repo_nwo" ]]; then
+      # Query GitHub API for fork parent (timeout prevents boot hang)
+      parent_nwo=$(timeout 10 gh repo view "$repo_nwo" --json isFork,parent \
+        --jq 'select(.isFork) | .parent.owner.login + "/" + .parent.name' \
+        2>/dev/null)
```
[the remark this answers is not in the document]
Comment (limbibot) [reply scripts/entrypoint.sh]: Not applicable — the container always clones via HTTPS (using `GH_PAT` for auth, line 216 of entrypoint.sh). There are no SSH keys in the container, so SSH remotes never appear. The upstream URL is also constructed as HTTPS explicitly (line 236).

File: approval/check-command.ts [hunk 2]
```diff
@@ -299,20 +398,35 @@ export async function isContextualGhCommand(command: string): Promise<boolean> {
   // Reject disallowed HTTP methods (anything outside GET/POST/PATCH allowlist)
   if (hasBlockedMethod(cmd)) return false;
 
-  // Parse target repo from the command
-  const target = parseGhApiTarget(cmd) ?? parseGhRepoFlag(cmd);
+  // Parse target repo from the command (including gh repo sync)
+  const syncResult = parseGhRepoSyncTarget(cmd);
+  const target =
+    parseGhApiTarget(cmd) ?? parseGhRepoFlag(cmd) ?? syncResult?.target ?? null;
```
[the remark this answers is not in the document]
Comment (limbibot) [reply approval/check-command.ts]: Fixed in 4cb921e. Scoped `hasBlockedMethod` to only fire when the command matches `^gh\s+api\b`. Added a test case for `gh issue comment` with `-X DELETE` in the body text — now correctly allowed.

**Review by perezd — APPROVED** [review]
I've run some manual smoke tests in a new instance and this feels stable to me.
